### ISSUE TITLE
Add book "chapter" markings and table of contents

### ISSUE THREAD

**Issue by jzohrab** [jzohrab/lute-v3#30]
It would be very nice to have a chapter marking or similar, and then have a table of contents or similar, showing the current chapter number at the top of the page perhaps.

When reading long texts, I sometimes want to know how many pages until I reach the end of the chapter.

**Comment by cblanken** [comment]
It seems like this might be a good candidate to integrate with the Bookmarks feature.

@jzohrab were you just imagining this as a separate slideout panel or more like a simple status component at the top that just listed the number of pages (and probably title) of the previous and next bookmarks?

**Comment by jzohrab** [comment]
Not sure, @cblanken -- it was a vague idea of perhaps a table of contents page.  The current bookmarks listing page would be totally sufficient.

As for showing the current chapter at the top of the page, that's not necessary.  I have an ereader, and it doesn't show the current chapter at the top, and I don't miss it.  To have the current chapter show at the top of the page, or to somehow indicate it in the page slider, we'd need to differentiate "chapter" bookmarks from regular bookmarks, and it doesn't feel worth it to me.

Ah "previous and next bookmarks" is kind of interesting, but also feels like overkill, given that people can have different types of bookmarks, e.g. topics that they want to chat with a tutor about.

Another vague idea I had was auto-creation of chapter markings ...  Maybe that would be another bookmark action -- "auto-generate bookmarks", e.g., and the user could enter some pattern indicating how a chapter is indicated (e.g., "line starts with 'Chapter'"), but that has so many exceptions and edge cases that it will definitely be more headache than it's worth.

**Comment by cblanken** [comment]
Automatic chapter markings definitely sounds like it'd be tough to make user friendly. You'd have to either give the user full control with a regex field or something, otherwise it'd turn into a huge form with a hundred options to select the right text.

It might be possible to setup automatic bookmark imports from epubs though. I think they store bookmarks in the same file. Not sure how it's stored though, so it might end up doing a search for the next `x` words/lines to find the right position after it converts everything into Lute pages which could be finicky.

**Comment by jzohrab** [comment]
Yeah finicky is the word here, and it could easily backfire and make a mess.  Maybe for imported text files where users can pre-edit the files to add special markers, that's ok, but for a general thing it's a drag.  Still a nice thing to backlog though (maybe :-P )

**Comment by jzohrab** [comment]
I kind of like the idea of using Markdown (ref #168) to general the table of contents, distinguishing it from bookmarks which are special user things.  This way, users could add "#" or "##" etc to the start of lines, and those would get converted into rich text, and a TOC could be generated from that _relatively_ easily.
